### PR TITLE
Backend login

### PR DESCRIPTION
**Pull request by dominikstraessle**

### CODE REVIEW

**Comment by mtte** [comment]
@dominikstraessle No the composer.lock file should always be checked in. You can read more about it [here](https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control).

File: api/composer.json [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+    "monolog/monolog": "^1.23",
+    "odan/slim-session": "^1.0",
+    "tuupola/slim-jwt-auth": "^3.1",
+    "vlucas/phpdotenv": "^2.5"
+  },
+  "autoload": {
+    "psr-4": {
+      "JMP\\": "src/JMP"
     }
+  }
```
Comment (mtte) [inline api/composer.json]: Add a script to automatically create the .env file from the example file.

```suggestion
  },
    "scripts": {
        "post-install-cmd": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ]
    }
```

File: README.md [hunk 2]
```diff
@@ -28,6 +28,11 @@ composer install -d api
 
 ### Docker
 
+Use the following command to start the environment for the first time:
+```bash
+docker-compose up -d --build
```
Comment (mtte) [inline README.md]: Is this really needed? As far as I know if started for the first time the containers are automatically built before they are created. Or am I missing something?

File: docs/api-v1.md [hunk 3]
```diff
@@ -10,9 +10,11 @@
     "username": "jake",
     "lastname": "Smith",
     "firstname": "Jacob",
+    "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE1NDM2ODUwMzYsImV4cCI6MTU0MzY5MjIzNiwianRpIjoiczlKSmErMGpPb2FGaWFxMTd6VHU5dz09IiwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdCIsInN1YiI6InNpbWkuZ3VsaSJ9.khH8KxrUWBco5eoiMHVgVRk2BMx1cMpvjDJH8j24sXs",
```
Comment (mtte) [inline docs/api-v1.md]: I do not think that the token should be here. Normally the token should only be returned when `/login` is called. So I suggest to change what is returned from the login route from a user object to just the token or the user object + token sepreatly. Therefore when retrieving a user (e.g. `/user`) the token is not exposed.

```json
{
    "token": "xxx",
    "user": {
        "id": 1,
        "username": "jake",
        "lastname": "Smith",
        "firstname": "Jacob",
        "email": "jake@example.com"
    }
}
```

```suggestion
```

File: README.md (hunk 2 above)
Comment (dominikstraessle) [reply README.md]: I'm not sure, we have to test this out.

File: docs/api-v1.md (hunk 3 above)
Comment (dominikstraessle) [reply docs/api-v1.md]: If we update this, we also have to adjust the login method in LoginController class

Comment (mtte) [reply docs/api-v1.md]: Ok, I updated it.

**Comment by mtte** [comment]
This is for now finished. We'll have to discuss the rest (comments, docker) later.

**Pull request merged**